cli/src/command.rs

```diff
@@ -267,7 +269,10 @@ fn run_node_inner(cli: Cli, overseer_gen: impl service::OverseerGen) -> Result<(
 
 /// Parses polkadot specific CLI arguments and run the service.
 pub fn run() -> Result<()> {
-	let cli = Cli::from_args();
+	let mut cli = Cli::from_args();
+	// Set the base path.
+	let base_path = BasePath::from_project("", "", &Cli::executable_name());
+	cli.run.base.shared_params.base_path = Some(base_path.path().to_owned());
```

Comment:
This should be moved into the match arm where it is needed, I assume we only need it for the host perf check right now?

Reply:
I think so, assigning it to the inner struct and extracting it back right away feels like a hack, unfortunately there's no easy way to set the path by default.
On the other hand, if some other subcommand will use it later, it's best to use the same path

Reply:
Yeah, this should not be done here. You also need to add the base-path to the HostPerfCheck cli command, so that the user can specify it.

Reply:
Do we need an option to specify it? This file is supposed to be something of an internal flag in order to
avoid running it multiple times. That is, if you run it for the first time with a custom path, you have to run it with this path specified every consequent use, otherwise this file will be saved in multiple locations.

If this subcommand is going to be executed at the start of the validator, do they have to care about another cli flag, even if it's optional?

And as a final downside, you have almost 0 chance to get an error while saving dotfile at the `BasePath` because it's in `HOME`.
If you specify `/` you may get `Permission denied`.

cc @drahnr